DigitalLearningSolutions.Data.Tests/Services/CourseAdminFieldsServiceTests.cs

```diff
@@ -0,0 +1,119 @@
+        }
+
+        [Test]
+        public void GetCustomPromptsForCourse_Returns_Populated_CourseAdminFields()
+        {
+            // Given
+            var expectedPrompt1 =
+                CustomPromptsTestHelper.GetDefaultCustomPrompt(1, "System Access Granted", "Yes\r\nNo");
+            var expectedPrompt2 = CustomPromptsTestHelper.GetDefaultCustomPrompt(2, "Access Permissions");
+            var customPrompts = new List<CustomPrompt> { expectedPrompt1, expectedPrompt2 };
+            var expectedCourseAdminFields = CustomPromptsTestHelper.GetDefaultCourseAdminFields(customPrompts);
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(27920, 101, 0))
+                .Returns(CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult());
+
+            // When
+            var result = courseAdminFieldsService.GetCustomPromptsForCourse(27920, 101, 0);
+
+            // Then
+            result.Should().BeEquivalentTo(expectedCourseAdminFields);
+        }
+
+        [Test]
+        public void GetCustomPromptsWithAnswersForCourse_Returns_Populated_List_of_CustomPromptWithAnswer()
+        {
+            // Given
+            const string answer1 = "ans1";
+            const string answer2 = "ans2";
+            var expected1 = CustomPromptsTestHelper.GetDefaultCustomPromptWithAnswer(
+                1,
+                "System Access Granted",
+                "Yes\r\nNo",
+                answer: answer1
+            );
+            var expected2 = CustomPromptsTestHelper.GetDefaultCustomPromptWithAnswer(
+                2,
+                "Access Permissions",
+                answer: answer2
+            );
+            var expected = new List<CustomPromptWithAnswer> { expected1, expected2 };
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(27920, 101, 0))
+                .Returns(CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult());
+            var delegateCourseInfo = new DelegateCourseInfo { Answer1 = answer1, Answer2 = answer2 };
+
+            // When
+            var result = courseAdminFieldsService.GetCustomPromptsWithAnswersForCourse(delegateCourseInfo, 27920, 101);
+
+            // Then
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Test]
+        public void UpdateCustomPromptForCourse_calls_data_service()
+        {
+            // Given
+            A.CallTo(() => courseAdminFieldsDataService.UpdateCustomPromptForCourse(1, 1, true, null)).DoesNothing();
+
+            // When
+            courseAdminFieldsService.UpdateCustomPromptForCourse(1, 1, true, null);
+
+            // Then
+            A.CallTo(() => courseAdminFieldsDataService.UpdateCustomPromptForCourse(1, 1, true, null)).MustHaveHappened();
+        }
+
+        [Test]
+        public void RemoveCustomPromptFromCourse_calls_data_service_with_correct_values()
+        {
+            using var transaction = new TransactionScope();
```

Comment:
Do we need transaction here if we mock the data services?

Reply:
Nope, that's a bit redundant - removed :)